src/components/dialogs/network-modifications/common/properties/property-utils.ts

```diff
@@ -0,0 +1,190 @@
+            return Promise.reject(
+                'Study entry could not be found in metadatas'
+            );
+        }
+
+        return studyMetadata[0].predefinedEquipmentProperties?.[
+            networkElementType
+        ]; // There should be only one study metadata
+    });
+};
+
+export const emptyProperties: Properties = {
+    [ADDITIONAL_PROPERTIES]: null,
+};
+
+export const initializedProperty = (): Property => {
+    return {
+        [NAME]: null,
+        [VALUE]: null,
+        [PREVIOUS_VALUE]: null,
+        [DELETION_MARK]: false,
+        [ADDED]: true,
+    };
+};
+
+export const getPropertiesFromModification = (
+    properties: Property[] | undefined
+): Properties => {
+    return {
+        [ADDITIONAL_PROPERTIES]: properties
+            ? properties.map((p) => {
+                  return {
+                      [NAME]: p[NAME],
+                      [VALUE]: p[VALUE],
+                      [PREVIOUS_VALUE]: null,
+                      [ADDED]: p[ADDED],
+                      [DELETION_MARK]: p[DELETION_MARK],
+                  };
+              })
+            : null,
+    };
+};
+
+export const getPropertiesFromEquipment = (
+    equipmentInfos: Equipment
+): Properties => {
+    return {
+        [ADDITIONAL_PROPERTIES]: equipmentInfos.properties
+            ? Object.entries(equipmentInfos.properties).map(([name, value]) => {
+                  return {
+                      [NAME]: name,
+                      [VALUE]: value,
+                      [PREVIOUS_VALUE]: null,
+                      [DELETION_MARK]: false,
+                      [ADDED]: true,
+                  };
+              })
+            : null,
+    };
+};
+
+export const concatProperties = (
+    modificationProperties: Property[],
+    equipment: Equipment
+): Property[] => {
+    const newModificationProperties = new Map<string, Property>();
+    for (const property of modificationProperties) {
+        if (property.name !== null) {
+            newModificationProperties.set(property.name, property);
+        }
+    }
+    if (equipment.properties !== undefined) {
+        Object.entries(equipment.properties).forEach(([name, value]) => {
+            if (name !== null) {
+                let propertyToAdd;
+                // If the property is present in the modification and in the equipment
+                if (newModificationProperties.has(name)) {
+                    const modProperty = newModificationProperties.get(name)!;
+                    propertyToAdd = {
+                        ...modProperty,
+                        previousValue: value, // We set previous value of the modification to the equipment value
+                    };
+                } else {
+                    propertyToAdd = {
+                        [NAME]: name,
+                        [VALUE]: null,
+                        [PREVIOUS_VALUE]: value,
+                        [DELETION_MARK]: false,
+                        [ADDED]: false,
```

Comment:
Here we create a newModificationProperty from an equipment property and [ADDED] is set to false. While in getPropertiesFromEquipment [ADDED] is set to true. I don't understand this difference. Does ADDED mean added to the modification hypothesis ?

Reply:
But I did the mistake previously, I thought we could always read "equipment properties" and return the same "modification properties" for both modification and creation dialogs but actually we couldn't.